packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,216 @@
+				// Test the memory usage of the SharedMatrix for removing a row in the middle for a given number of times.
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Remove a row in the middle ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.removeRows(Math.floor(size / 2), 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix(
+								"testLocalMatrix",
+								size + count,
+								matrixValue,
+							);
+						}
+					})(),
+				);
+
+				// Test the memory usage of the SharedMatrix for inserting a row and a column in the middle for a given number of times.
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Insert a row and a column ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertCols(Math.floor(size / 2), 1);
+								this.localMatrix.insertRows(Math.floor(size / 2), 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix("testLocalMatrix", size, matrixValue);
+						}
+					})(),
+				);
+
+				// Test the memory usage of the SharedMatrix for removing a row and a column in the middle for a given number of times.
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Remove a row and a column ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.removeCols(Math.floor(size / 2), 1);
+								this.localMatrix.removeRows(Math.floor(size / 2), 1);
+							}
+						}
+
+						beforeIteration(): void {
+							// Make sure the matrix is large enough to remove rows from it. At least 2*2
+							this.localMatrix = createLocalMatrix(
+								"testLocalMatrix",
+								size + count + 2,
+								matrixValue,
+							);
+						}
+					})(),
+				);
+
+				// Test the memory usage of the SharedMatrix for setting a string in a cell for a given number of times.
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Set a 3-character string in ${count} cells`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.setCell(Math.floor(size / 2), i, "abc");
+							}
```

Comment:
As this is currently written, we will only modify cells of the middle row of the table for each column. I think we should be a bit more sporadic than that, to ensure we don't miss implications of column _and_ row differences.

An easy way to do this might be to use `setCell(i, i, "abc")` - that way we're changing the row and column each time.

Reply:
Make sense. Updated